src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -199,6 +200,32 @@ class spell_warr_colossus_smash : public SpellScript
     }
 };
 
+// 389306 - Critical Thinking
+class spell_warr_critical_thinking : public AuraScript
+{
+    PrepareAuraScript(spell_warr_critical_thinking);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_CRITICAL_THINKING_ENERGIZE });
+    }
+
+    void HandleProc(AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        Unit* target = GetTarget();
+        int32 rageAmount = *eventInfo.GetProcSpell()->GetPowerTypeCostAmount(POWER_RAGE);
```

Comment:
GetPowerTypeCostAmount returns a Optional, you cannot simply access its value without checking if it has a value